### ISSUE TITLE
Latest changes (0.5.0) break the S3BotoStorage's use of location

### ISSUE THREAD

**Issue by johnyoonh** [antonagestam/collectfast#91]
I use the follow code so that I could set the prefix of the location to be the same as the one on the local directory:
```
from django.conf import settings
from storages.backends.s3boto import S3BotoStorage

class StaticStorage(S3BotoStorage):
    location = settings.STATICFILES_LOCATION
    def path(self, name):
        return self.url(name)

class MediaStorage(S3BotoStorage):
    location = settings.MEDIAFILES_LOCATION
    def path(self, name):
        return self.url(name)
~
```
Prior to 0.5.0, the `location` will only affect the result directory on s3, but now it tries to look for the static files while taking this variable into account.

For example. If STATICFILES_LOCATION = "static"
In the past, the search path would have been "/static/js/webpack.js"
Now it is "/static/static/js/webpack.js"

this variable `location` should affect where Collectfast is looking for files. This happens not only to the files under /static but for other static files, such as admin related things. There is an extra "/static" in the path.
This was reproduced in 0.5.1 too

**Comment by antonagestam** [comment]
This should have been fixed in 0.5.1. I can reproduce this in 0.5.0 using `AWS_LOCATION = "/static"` but not in 0.5.0. See https://github.com/antonagestam/collectfast/pull/90

Is there a reason you're overriding the path method?

Btw, I recommend upgrading to boto3 and using the S3Boto3Storage.

**Comment by antonagestam** [comment]
Closing this until further information is supplied.
